src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -361,6 +361,70 @@ def update_ag_http_listener(cmd, instance, parent, item_name, frontend_ip=None,
     return parent
 
 
+def assign_ag_identity(cmd, resource_group_name, application_gateway_name,
+                       no_wait=False, identity_type=None, user_assigned_identity=None,
+                       client_id=None, principal_id=None):
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    ManagedServiceIdentity, ManagedServiceIdentityUserAssignedIdentitiesValue = \
+        cmd.get_models('ManagedServiceIdentity', 'ManagedServiceIdentityUserAssignedIdentitiesValue')
+    user_assigned_indentity_instance = ManagedServiceIdentityUserAssignedIdentitiesValue(
+        client_id=client_id,
+        principal_id=principal_id
+    )
+    user_assigned_identities_instance = dict()
+    if user_assigned_identity is not None:
+        user_assigned_identities_instance[user_assigned_identity] = user_assigned_indentity_instance
+    else:
+        user_assigned_identities_instance = None
+    identity_instance = ManagedServiceIdentity(
+        type=identity_type,
+        user_assigned_identities=user_assigned_identities_instance
+    )
+    ag.identity = identity_instance
+
+    return sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag)
+
+
+def add_ag_identity(cmd, resource_group_name, application_gateway_name,
```

Comment:
This function isn't used?

Reply:
yes. I add comments in the next commit. I will push the latest one.

Reply:
Why do you add unused function here?

Reply:
Well. it may be used in future. What's your concern about this unused function?